### ISSUE TITLE
Implement Discount Codes API with attributes, relationships and permissions

### ISSUE THREAD

**Issue by enigmaeth** [fossasia/open-event-orga-server#3735]
To be done after #3728, #3720.

**Comment by enigmaeth** [comment]
@SaptakS @shubham-padia @poush
What permissions should be given for the discount_code API? 
Detail : 
Get: jwt_required 
Others: is_organizer

List : is_organizer

Rationship: is_organizer

**Comment by poush** [comment]
@enigmaeth These decorators looks fine. I think it is not required to put decorator on relationships because eventually the other endpoint is protected with its own rules and decorators.

If #3726 get merged before feel free to add them or you can make new PR once it gets merged.

**Comment by poush** [comment]
and yeah I think it should be @co-organizer instead of organizer

**Comment by enigmaeth** [comment]
Closed via #3776.
